monitoring/notification/dashboard.py

```diff
@@ -0,0 +1,516 @@
+    legendPlacement='right',
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='\n'.join([
+            'label_replace(',
+            'sum by(job) (rate(' + Metrics.CACHE_LAG.sum(cache_hit="true") + '))',
+            ', "job", "$1", "job", "' + LABEL_EXPR['notification-processors'] +'")',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+notification_delivery_delay = Heatmap(
+    title='Delivery Delay',
+    description='Delay between when a notification is sent and when it is received',
+    dataSource='${DS_PROMETHEUS}',
+    hideZeroBuckets=True,
+    tooltip=Tooltip(show=True, showHistogram=True),
+    yAxis=YAxis(format=UNITS.SECONDS, decimals=0),
+    cards={'cardPadding': 1, 'cardRound': 2},
+    color=HeatmapColor(mode='opacity'),
+    targets=[Target(
+        expr='sum(rate(' + Metrics.DELIVERY_LAG.sum(status="success") + ')) by(le)',
+        legendFormat='{{le}}',
+    )])
+
+cached_buckets = TimeSeries(
+    title='Cached Notification Configurations',
+    description='Total number of notification configurations cached',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    targets=[Target(
+        expr='\n'.join([
+            'label_replace(',
+            'sum(' + Metrics.CACHED_BUCKETS.raw()  + ') by (job)',
+            ', "job", "$1", "job", "' + LABEL_EXPR['notification-processors'] +'")',
+        ]),
+        legendFormat='{{job}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+cache_updates = TimeSeries(
+    title='Cache Updates',
+    description='Rate of cache updates',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='\n'.join([
+            'label_replace(',
+            'sum by (job) (rate (' + Metrics.CACHE_UPDATES() + '))',
+            ', "job", "$1", "job", "' + LABEL_EXPR['notification-processors'] +'")',
+        ]),
+        legendFormat='{{job}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+cache_hit_miss_ratio = GaugePanel(
+    title='Cache Hit/Miss Ratio',
+    description='Cache Hit/Miss Ratio',
+    dataSource='${DS_PROMETHEUS}',
+    noValue='0',
+    min='0',
+    max='1',
+    format=UNITS.PERCENT_UNIT,
+    targets=[Target(
+        expr='\n'.join([
+            'label_replace(',
+            'sum by (job) (rate (' + Metrics.CACHE_LAG.count(cache_hit="true") + '))',
+            '/',
+            'sum by (job) (rate (' + Metrics.CACHE_LAG.count() +'))',
+            ', "job", "$1", "job", "' + LABEL_EXPR['notification-processors'] +'")',
+        ]),
+        legendFormat='{{job}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
```

Comment:
I am wondering if we could stick to common standards heres, i.e. a red color if the rate is below 80%, as this is considered a poor cache hit rate, something like
```
thresholds=[
        Threshold("red",     1, 0.00),
        Threshold("orange",  2, 0.85),
        Threshold("green",   3, 0.90),
    ],
```

Mixed with more info in the gauge to ease the reading, like 
![image](https://user-images.githubusercontent.com/91462779/200856483-040d6ee4-615e-43c9-b038-cb0333deee5d.png)

Reply:
agreed